File: array_problems/sortArray/sortArray.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class ArrayProblems
{
    private int[] arr;

    public ArrayProblems(int[] arr)
    {
        this.arr = (int[])arr.Clone(); 
    }

    public int[] Quick(int[] items = null)
    {
        int[] list = items == null ? (int[])arr.Clone() : (int[])items.Clone();
        if (list.Length < 2) return list;
        int pivot = list[0];
        List<int> smaller = list.Skip(1).Where(item => item < pivot).ToList();
        List<int> bigger = list.Skip(1).Where(item => item > pivot).ToList();
        List<int> equal = list.Where(item => item == pivot).ToList();
        return Quick(smaller.ToArray()).Concat(equal.ToArray()).Concat(Quick(bigger.ToArray())).ToArray();
    }

    public int[] Bubble(int[] items = null)
    {
        int[] list = items == null ? (int[])arr.Clone() : (int[])items.Clone();
        int length = list.Length;
        for (int passover = 0; passover < length; passover++)
        {
            for (int index = 0; index < length - 1; index++)
            {
                if (list[index] > list[index + 1])
                {
                    int temporary = list[index];
                    list[index] = list[index + 1];
                    list[index + 1] = temporary;
                }
            }
        }
        return list;
    }

    public int[] Select(int[] items = null)
    {
        int[] list = items == null ? (int[])arr.Clone() : (int[])items.Clone();
        int length = list.Length;
        for (int passes = 0; passes < length; passes++)
        {
            int min = passes;
            for (int i = passes; i < length; i++)
            {
                if (list[i] < list[min]) min = i;
            }
            if (min != passes)
            {
                int temporary = list[passes];
                list[passes] = list[min];
                list[min] = temporary;
            }
        }
        return list;
    }

    public int[] Insert(int[] items = null)
    {
        int[] list = items == null ? (int[])arr.Clone() : (int[])items.Clone();
        int length = list.Length;
        for (int i = 1; i < length; i++)
        {
            int index = i - 1;
            int temporary = list[i];
            while (index >= 0 && list[index] > temporary)
            {
                list[index + 1] = list[index];
                index--;
            }
            list[index + 1] = temporary;
        }
        return list;
    }

    public int[] Simple(int[] items = null)
    {
        int[] list = items == null ? (int[])arr.Clone() : (int[])items.Clone();
        int length = list.Length;
        for (int i = 0; i < length; i++)
        {
            for (int j = 0; j < length; j++)
            {
                if (list[i] < list[j])
                {
                    int temp = list[i];
                    list[i] = list[j];
                    list[j] = temp;
                }
            }
        }
        return list;
    }

    public int[] MergeSort(int[] items = null)
    {
        int[] list = items == null ? (int[])arr.Clone() : (int[])items.Clone();
        if (list.Length <= 1) return list;
        int middle = list.Length / 2;
        int[] left = MergeSort(list.Take(middle).ToArray());
        int[] right = MergeSort(list.Skip(middle).ToArray());
        return Merge(left, right);
    }

    private int[] Merge(int[] left, int[] right)
    {
        List<int> merged = new List<int>();
        int leftIndex = 0;
        int rightIndex = 0;
        while (leftIndex < left.Length && rightIndex < right.Length)
        {
            if (left[leftIndex] <= right[rightIndex])
            {
                merged.Add(left[leftIndex++]);
            }
            else
            {
                merged.Add(right[rightIndex++]);
            }
        }
        while (leftIndex < left.Length) merged.Add(left[leftIndex++]);
        while (rightIndex < right.Length) merged.Add(right[rightIndex++]);
        return merged.ToArray();
    }
}

public class Program
{
    public static void Main(string[] args)
    {
        int[] arr = { 0, 43, 3, 2, 3, 4, 6 };
        ArrayProblems sort = new ArrayProblems(arr);

        int[] quickSort = sort.Quick();
        int[] selectSort = sort.Select();
        int[] insertSort = sort.Insert();
        int[] bubbleSort = sort.Bubble();
        int[] simpleSort = sort.Simple();
        int[] mergeSort = sort.MergeSort();

        Console.WriteLine("quick sort: " + string.Join(", ", quickSort));
        Console.WriteLine("select sort: " + string.Join(", ", selectSort));
        Console.WriteLine("insert sort: " + string.Join(", ", insertSort));
        Console.WriteLine("bubble sort: " + string.Join(", ", bubbleSort));
        Console.WriteLine("simple sort: " + string.Join(", ", simpleSort));
        Console.WriteLine("merge sort: " + string.Join(", ", mergeSort));
    }
}

File: array_problems/sortArray/sortArray.php
<?php
class ArrayProblems {
    private $arr;
    public function __construct($arr) {
        $this->arr = $arr;
    }

    public function quick($items = null) {
        $list = $items ?? $this->arr;
        if (count($list) < 2) return $list;
        $pivot = $list[0];
        $smaller = array_filter(array_slice($list, 1), function($item) use ($pivot) {
            return $item < $pivot;
        });
        $bigger = array_filter(array_slice($list, 1), function($item) use ($pivot) {
            return $item > $pivot;
        });
        $equal = array_filter($list, function($item) use ($pivot) {
            return $item == $pivot;
        });
        return array_merge($this->quick(array_values($smaller)), array_values($equal), $this->quick(array_values($bigger))); //Corrected: array_values to reindex.
    }

    public function bubble($items = null) {
        $items = $items ?? $this->arr;
        $length = count($items);
        for ($passover = 0; $passover < $length; $passover++) {
            for ($index = 0; $index < $length - 1; $index++) {
                if ($items[$index] > $items[$index + 1]) {
                    $temporary = $items[$index];
                    $items[$index] = $items[$index + 1];
                    $items[$index + 1] = $temporary;
                }
            }
        }
        return $items;
    }

    public function select($items = null) {
        $items = $items ?? $this->arr;
        $length = count($items);
        for ($passes = 0; $passes < $length; $passes++) {
            $min = $passes;
            for ($i = $passes; $i < $length; $i++) {
                if ($items[$i] < $items[$min]) $min = $i;
            }
            if ($min !== $passes) {
                $temporary = $items[$passes];
                $items[$passes] = $items[$min];
                $items[$min] = $temporary;
            }
        }
        return $items;
    }

    public function insert($items = null) {
        $items = $items ?? $this->arr;
        $length = count($items);
        for ($i = 1; $i < $length; $i++) {
            $index = $i - 1;
            $temporary = $items[$i];
            while ($index >= 0 && $items[$index] > $temporary) {
                $items[$index + 1] = $items[$index];
                $index--;
            }
            $items[$index + 1] = $temporary;
        }
        return $items;
    }

    public function simple($items = null) {
        $items = $items ?? $this->arr;
        $length = count($items);
        for ($i = 0; $i < $length; $i++) {
            for ($j = 0; $j < $length; $j++) {
                if ($items[$i] < $items[$j]) {
                    $temp = $items[$i];
                    $items[$i] = $items[$j];
                    $items[$j] = $temp;
                }
            }
        }
        return $items;
    }

    public function mergeSort($items = null) {
        $items = $items ?? $this->arr;
        if (count($items) <= 1) return $items;
        $middle = floor(count($items) / 2);
        $left = $this->mergeSort(array_slice($items, 0, $middle));
        $right = $this->mergeSort(array_slice($items, $middle));
        return $this->merge($left, $right);
    }

    private function merge($left, $right) {
        $merged = [];
        while (count($left) > 0 && count($right) > 0) {
          ($left[0] <= $right[0]) ? $merged[] = array_shift($left) : $merged[] = array_shift($right);
        }
        return array_merge($merged, $left, $right);
    }
}

$arr = [0, 43, 3, 2, 3, 4, 6];
$sort = new ArrayProblems($arr);

$quickSort = $sort->quick(); 
$selectSort = $sort->select(); 
$insertSort = $sort->insert(); 
$bubbleSort = $sort->bubble(); 
$simpleSort = $sort->simple(); 
$mergeSort = $sort->mergeSort(); 

echo "quick sort: " . implode(", ", $quickSort) . "\n";
echo "select sort: " . implode(", ", $selectSort) . "\n";
echo "insert sort: " . implode(", ", $insertSort) . "\n";
echo "bubble sort: " . implode(", ", $bubbleSort) . "\n";
echo "simple sort: " . implode(", ", $simpleSort) . "\n";
echo "merge sort: " . implode(", ", $mergeSort) . "\n";
?>